File: Code/Chapter 9/VB.NET/Exercise 9.1/Form1.vb
Public Class Form1

   Private Sub bindingNavigatorSaveItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bindingNavigatorSaveItem.Click
      If Me.Validate Then
         Me.AnimalsBindingSource.EndEdit()
         Me.AnimalsTableAdapter.Update(Me.TestDataSet.Animals)
      Else
         System.Windows.Forms.MessageBox.Show(Me, "Validation errors occurred.", "Save", System.Windows.Forms.MessageBoxButtons.OK, System.Windows.Forms.MessageBoxIcon.Warning)
      End If

   End Sub

   Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
      'TODO: This line of code loads data into the 'TestDataSet.Animals' table. You can move, or remove it, as needed.
      Me.AnimalsTableAdapter.Fill(Me.TestDataSet.Animals)

   End Sub
End Class

File: Code/Chapter 9/C#/Exercise 9.1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Exercise_9_1
{
	public partial class Form1: Form
	{
		public Form1()
		{
			InitializeComponent();
		}

      private void bindingNavigatorSaveItem_Click(object sender, EventArgs e)
      {
if (this.Validate()) {
    this.animalsBindingSource.EndEdit();
    this.animalsTableAdapter.Update(this.testDataSet.Animals);
}
else {
    System.Windows.Forms.MessageBox.Show(this, "Validation errors occurred.", "Save", System.Windows.Forms.MessageBoxButtons.OK, System.Windows.Forms.MessageBoxIcon.Warning);
}
      
      }

      private void Form1_Load(object sender, EventArgs e)
      {
// TODO: This line of code loads data into the 'testDataSet.Animals' table. You can move, or remove it, as needed.
this.animalsTableAdapter.Fill(this.testDataSet.Animals);
      
      }    
	}
}